### PR TITLE
Add coap adapter deployment

### PR DESCRIPTION
**Pull request by boaks**
Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: deploy/src/main/deploy/example-credentials.json [hunk 1]
```diff
@@ -40,6 +40,26 @@
...
+            "not-after": "2037-06-01T14:00:00+01:00",
+            "comment": "coap preshared key: coap-secret",
+            "key" : "Y29hcC1zZWNyZXQ="
+          }
+        ]
+      },
+      {
+        "device-id": "4711",
+        "type": "x509-cert",
+        "auth-id": "C=CA,L=Ottawa,O=Eclipse IoT,OU=Californium,CN=cf-client",
```
Comment (sophokles73) [inline deploy/src/main/deploy/example-credentials.json]: this doesn't seem to be related to PSK auth, is it?

File: deploy/src/main/deploy/example-tenants.json [hunk 2]
```diff
@@ -1,7 +1,12 @@
 [
   {
     "tenant-id": "DEFAULT_TENANT",
-    "enabled": true
+    "enabled": true,
+    "trusted-ca": {
+      "subject-dn": "C=CA,L=Ottawa,O=Eclipse IoT,OU=Californium,CN=cf-ca",
+      "type": "EC",
+      "public-key": "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEGTYGKFKF3grlWqJFBYkiLGMT3tTSyCGf/IoalXoCx38N2FaB6j6cVJmYlQZecYtnQNh+BgEfK8OLv3n8zwFP2Q=="
```
Comment (sophokles73) [inline deploy/src/main/deploy/example-tenants.json]: this also doesn't seem to be related to PSK auth

File: deploy/src/main/deploy/docker/swarm_deploy.sh [hunk 3]
```diff
@@ -19,14 +19,27 @@ CONFIG=$SCRIPTPATH/../../config
...
+  SLIM=0
+else 
+  SLIM=$1
+fi
 
+if [ $SLIM -ne 1 ]
+then
 echo DEPLOYING ECLIPSE HONO TO DOCKER SWARM
+else
+echo SLIM DEPLOYING ECLIPSE HONO TO DOCKER SWARM
```
Comment (sophokles73) [inline deploy/src/main/deploy/docker/swarm_deploy.sh]: what's the purpose of this?

File: deploy/src/main/deploy/example-credentials.json (hunk 1 above)
Comment (boaks) [reply deploy/src/main/deploy/example-credentials.json]: Sure, I remove it.

File: deploy/src/main/deploy/example-tenants.json (hunk 2 above)
Comment (boaks) [reply deploy/src/main/deploy/example-tenants.json]: I remove it.

File: deploy/src/main/deploy/docker/swarm_deploy.sh (hunk 3 above)
Comment (boaks) [reply deploy/src/main/deploy/docker/swarm_deploy.sh]: Commit comment:

> Add option for slim deployment, which omit to deploy the adapters in
order to start an adapter from a IDE.

It makes it easier to develop an adapter. It starts all required parts except the adapter, which is the started separately. Enables faster turn-around during adapter development.

File: deploy/src/main/deploy/docker/swarm_deploy.sh [hunk 4]
```diff
@@ -19,14 +19,27 @@ CONFIG=$SCRIPTPATH/../../config
 CERTS=$CONFIG/hono-demo-certs-jar
 NS=hono
 CREATE_OPTIONS="-l project=$NS --network $NS --detach=false"
+if [ -z "$1" ] ; then
```
Comment (sophokles73) [inline deploy/src/main/deploy/docker/swarm_deploy.sh]: can we do a check for something meaningful, e.g. `--no-adapters`?

File: deploy/src/main/deploy/docker/swarm_deploy.sh [hunk 5]
```diff
@@ -78,7 +93,7 @@ echo Deploying Qpid Dispatch Router ...
 docker secret create -l project=$NS qdrouter-key.pem $CERTS/qdrouter-key.pem
 docker secret create -l project=$NS qdrouter-cert.pem $CERTS/qdrouter-cert.pem
 docker secret create -l project=$NS qdrouterd.json $SCRIPTPATH/qpid/qdrouterd-with-broker.json
-docker service create $CREATE_OPTIONS --name hono-dispatch-router -p 15671:5671 -p 15672:5672 \
+docker service create $CREATE_OPTIONS --name hono-dispatch-router -p 15671:5671 -p 15672:5672 -p 15673:5673 \
```
Comment (sophokles73) [inline deploy/src/main/deploy/docker/swarm_deploy.sh]: I would rather have this port exposed only if `--no-adapters` is set ...

File: deploy/src/main/deploy/docker/swarm_deploy.sh (hunk 4 above)
Comment (boaks) [reply deploy/src/main/deploy/docker/swarm_deploy.sh]: Sure, if you also think it makes sense to have a "no adapter" deployment, then we can adapt this idea.

File: deploy/src/main/deploy/docker/swarm_deploy.sh (hunk 5 above)
Comment (boaks) [reply deploy/src/main/deploy/docker/swarm_deploy.sh]: Sure.

**Comment by boaks** [comment]
Changed deploy switch to "--no-adapter" and removed the exposing of "-p 15673:5673" from full deployment.

File: demo-certs/certs/coap-adapter-cert.pem [hunk 6]
```diff
@@ -0,0 +1,64 @@
+-----BEGIN CERTIFICATE-----
```
Comment (sophokles73) [inline demo-certs/certs/coap-adapter-cert.pem]: has this cert been added to the existing certs, i.e. has it been created/signed using the *existing* intermediate cert or have you re-created *all* keys and certificates by running the full create_certs script?

Comment (boaks) [reply demo-certs/certs/coap-adapter-cert.pem]: I hope, I used the existing certs :-).
At least, using this certs and to already available other certs, allows to connect the coap-adapter to the qpid-router.
